core/sql/optimizer/SimpleScanOptimizer.cpp

```diff
@@ -218,25 +218,14 @@ SimpleFileScanOptimizer::isLogicalSubPartitioned() const
 SearchKey *
 SimpleFileScanOptimizer::constructSearchKey()
 {
+  // We do not need to incude the flatten version of RANGE SPEC predicates 
+  // to the exePreds because SearchKey is capable of handling such predicates.
```

Comment:
Good question, and it makes me think of two things. One is that the example where MDAM can't retrieve the exact set is probably one that involves complex predicates on two or more columns, while RangeSpec itself is probably simple enough. But RangeSpec AND <some other predicate> might fall into this category. The second question is whether we really handle RangeSpec predicates correctly in MDAM code generation.